### PR TITLE
Fix devicePixelRatio check

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: src/library_html5.js [hunk 1]
```diff
@@ -3181,11 +3181,11 @@ var LibraryJSEvents = {
   emscripten_get_device_pixel_ratio__proxy: 'sync',
   emscripten_get_device_pixel_ratio__sig: 'd',
   emscripten_get_device_pixel_ratio: function() {
+// https://caniuse.com/#search=devicePixelRatio
 #if ENVIRONMENT == 'web' && MIN_IE_VERSION > 11 && MIN_FIREFOX_VERSION >= 18
-    // Save a little bit of code space: all Wasm-capable browsers support devicePixelRatio.
     return devicePixelRatio;
 #else
-    return devicePixelRatio || 1.0;
+     return (typeof devicePixelRatio === 'number' && devicePixelRatio) || 1.0;
```
Comment (juj) [inline src/library_html5.js]: Ohh, very good catch! It looks like I regressed this in https://github.com/emscripten-core/emscripten/commit/2ca4386c067ea83e82e320f07db7e4d5961c596a#diff-291d81f9d42b322a89881b6d91f7a122L2994 .

Let's revert back to the old form?
```js
return window.devicePixelRatio || 1.0;
```

Comment (kripken) [reply src/library_html5.js]: In the if or the else?

In the if, isn't just `devicePixelRatio` ok already, as it's written?

In the else, we need to handle the case of non-web (like worker or node.js), so we can't use `window.`?

**Comment by kripken** [comment]
Btw, this should probably block tagging a new version, as without this we can crash on web workers.

File: src/library_html5.js (hunk 1 above)
Comment (juj) [reply src/library_html5.js]: Let's go with

```js
  emscripten_get_device_pixel_ratio__proxy: 'sync',
  emscripten_get_device_pixel_ratio__sig: 'd',
  emscripten_get_device_pixel_ratio: function() {
#if MIN_IE_VERSION < 11 || MIN_FIREFOX_VERSION < 18 || MIN_CHROME_VERSION < 4 || MIN_SAFARI_VERSION < 3010 // https://caniuse.com/#feat=devicepixelratio
    return window.devicePixelRatio || 1.0;
#else
    return devicePixelRatio;
#endif
  }
```
if that looks good?

Yeah, web workers don't have `window`, and there is neither a `WorkerGlobalScope.devicePixelRatio` variable. That is why `emscripten_get_device_pixel_ratio()` is implemented as a sync proxied function, so this function is never executed in web worker context.

Comment (kripken) [reply src/library_html5.js]: Oh right, thanks, it's proxied... I missed that.

Modified to how you suggested.

Comment (kripken) [reply src/library_html5.js]: Oh, but we also need the fix for node.js. Added that too.

(Some people run tests on node.js which include calls like this; it's nice not to crash there.)

**Pull request merged**